src/js_modules/profile/Subscriptions/ButtonHandler.jsx

```diff
@@ -3,40 +3,51 @@ import { Button } from '@chakra-ui/react';
 import useTranslation from 'next-translate/useTranslation';
 import PropTypes from 'prop-types';
 import { useState } from 'react';
-import Link from '../../../common/components/NextChakraLink';
 import profileHandlers from './handlers';
 
 function ButtonHandler({
-  subscription, onOpenUpgrade, setSubscriptionProps, onOpenCancelSubscription, children, ...restStyles
+  subscription, onOpenUpgrade, setSubscriptionProps, onOpenCancelSubscription, children, allSubscriptions, ...restStyles
 }) {
   const { t } = useTranslation('profile');
   const status = subscription?.status;
   const [isLoading, setIsLoading] = useState(false);
   const isFullyPaid = subscription?.status?.toLowerCase() === 'fully_paid';
   const planSlug = subscription?.plans?.[0]?.slug;
   const isPlanFinancingExpired = subscription?.type === 'plan_financing' && subscription?.valid_until < new Date().toISOString();
+  const planOfferedAcquired = allSubscriptions?.some((sub) => sub.plans.some((plan) => plan.slug === subscription?.planOffer?.slug));
 
-  const {
-    getPlanOffer,
-  } = profileHandlers({});
-
+  const { getPlanOffer, reactivatePlan } = profileHandlers({});
   const handlePlanOffer = () => {
     setIsLoading(true);
-    getPlanOffer({ slug: planSlug, onOpenUpgrade })
-      .finally(() => setIsLoading(false));
+    getPlanOffer({ slug: planSlug, onOpenUpgrade }).finally(() => setIsLoading(false));
+  };
+
+  const handleReactivatePlan = () => {
+    setIsLoading(true);
+    reactivatePlan(planSlug, status);
   };
 
   const getStyles = () => {
-    if (subscription?.type !== 'plan_financing' && (status === 'ACTIVE' || status === 'FULLY_PAID')) {
+    if (planOfferedAcquired && (status === 'FREE_TRIAL' || status === 'FULLY_PAID')) {
       return {
-        text: t('subscription.cancel'),
+        text: '',
         style: {
-          variant: 'link',
+          display: 'none',
         },
       };
     }
 
-    if (status === 'FREE_TRIAL' || (isPlanFinancingExpired && subscription?.planOffer?.pricing_exists)) {
+    if (planOfferedAcquired && status === 'ACTIVE') {
+      return {
+        text: '',
+        style: {
+          display: 'none',
+        },
+      };
+    }
+
+    if (status === 'FREE_TRIAL' || (isPlanFinancingExpired && subscription?.planOffer?.pricing_exists)
```

Comment:
It's redundant, if status is FREE_TRIAL, then the first condition is ignored or will it apply both styles?